sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Verifier.cs

```diff
@@ -0,0 +1,679 @@
+            }
+        }
+
+        private static void VerifyFixAllCodeFix(string path, string pathToExpected, DiagnosticAnalyzer diagnosticAnalyzer,
+            CodeFixProvider codeFixProvider, string codeFixTitle, params MetadataReference[] additionalReferences)
+        {
+            var fixAllProvider = codeFixProvider.GetFixAllProvider();
+            if (fixAllProvider == null)
+            {
+                return;
+            }
+
+            using (var workspace = new AdhocWorkspace())
+            {
+                var file = new FileInfo(path);
+                var parseOptions = GetParseOptionsWithDifferentLanguageVersions(null, file.Extension);
+
+                foreach (var parseOption in parseOptions)
+                {
+                    var document = CreateProject(file.Extension, GeneratedAssemblyName, workspace, additionalReferences)
+                        .AddDocument(file, true)
+                        .Documents
+                        .Single(d => d.Name == file.Name);
+                    RunFixAllProvider(diagnosticAnalyzer, codeFixProvider, codeFixTitle, fixAllProvider, document, parseOption, pathToExpected);
+                }
+            }
+        }
+
+        private static Project CreateProject(string fileExtension, string assemblyName,
+            AdhocWorkspace workspace, params MetadataReference[] additionalReferences)
+        {
+            var language = fileExtension == CSharpFileExtension
+                ? LanguageNames.CSharp
+                : LanguageNames.VisualBasic;
+
+            var project = workspace.CurrentSolution.AddProject(assemblyName, $"{assemblyName}.dll", language)
+                .AddMetadataReference(systemAssembly)
+                .AddMetadataReference(systemLinqAssembly)
+                .AddMetadataReference(systemNetAssembly)
+                .AddMetadataReferences(additionalReferences);
+
+            // adding an extra file to the project
+            // this won't trigger any issues, but it keeps a reference to the original ParseOption, so
+            // if an analyzer/codefix changes the language version, Roslyn throws an ArgumentException
+            project = project.AddDocument("ExtraEmptyFile.g" + fileExtension, string.Empty).Project;
+
+            return project;
+        }
+
+        private static Project AddDocument(this Project project, FileInfo file,
+            bool removeAnalysisComments = false)
+        {
+            var text = ReadDocument(file.FullName, removeAnalysisComments);
+
+            return project.AddDocument(file.Name, text).Project;
+        }
+
+        private static string ReadDocument(string filePath, bool removeAnalysisComments)
+        {
+            var lines = File.ReadAllText(filePath, Encoding.UTF8)
+                .Replace(WindowsLineEnding, UnixLineEnding) // This allows to deal with multiple line endings
+                .Split(new[] { UnixLineEnding }, StringSplitOptions.None);
```

Comment:
I have changed this code to convert all line endings to unix then split on it. There was some bug on the CI that was reading some files as LF while the code was assuming CRLF.